app/controllers/autocomplete_controller.rb

```diff
@@ -103,7 +104,34 @@ def sort_suggestions(suggestions)
 
   URI_PREFIX = 'http://id.worldcat.org/fast/'
 
-  def authority_uri(idroot)
-    "#{URI_PREFIX}#{idroot.delete_prefix('fst').to_i}/"
+  def key(idroot, type)
+    "#{URI_PREFIX}#{idroot.delete_prefix('fst').to_i}/::#{cocina_type(type)}"
   end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/MethodLength
+  def cocina_type(type)
```

Comment:
Can this be a Hash rather than a case statement?

Reply:
Yup. Fixed.